src/networkextension.cs

```diff
@@ -2104,4 +2134,144 @@ interface NETransparentProxyNetworkSettings {
 		[NullAllowed, Export ("excludedNetworkRules", ArgumentSemantic.Copy)]
 		NENetworkRule[] ExcludedNetworkRules { get; set; }
 	}
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[BaseType (typeof (NSObject))]
+	interface NEAppPushManager {
+		[Export ("matchSSIDs", ArgumentSemantic.Copy)]
+		string[] MatchSsids { get; set; }
+
+		[Export ("providerConfiguration", ArgumentSemantic.Copy)]
+		NSDictionary<NSString, NSObject> ProviderConfiguration { get; set; }
+
+		[NullAllowed]
+		[Export ("providerBundleIdentifier")]
+		string ProviderBundleIdentifier { get; set; }
+
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		INEAppPushDelegate Delegate { get; set; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Weak)]
+		NSObject WeakDelegate { get; set; }
+
+		[Async]
+		[Static]
+		[Export ("loadAllFromPreferencesWithCompletionHandler:")]
+		void LoadAllFromPreferences (Action<NEAppPushManager [], NSError> completionHandler);
+
+		[Async]
+		[Export ("loadFromPreferencesWithCompletionHandler:")]
+		void LoadFromPreferences (Action<NSError> completionHandler);
+
+		[Async]
+		[Export ("removeFromPreferencesWithCompletionHandler:")]
+		void RemoveFromPreferences (Action<NSError> completionHandler);
+
+		[Async]
+		[Export ("saveToPreferencesWithCompletionHandler:")]
+		void SaveToPreferences (Action<NSError> completionHandler);
+
+		[NullAllowed]
+		[Export ("localizedDescription")]
+		string LocalizedDescription { get; set; }
+
+		[Export ("enabled")]
+		bool Enabled { [Bind ("isEnabled")] get; set; }
+
+		[Export ("active")]
+		bool Active { [Bind ("isActive")] get; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (14,0)]
+	[BaseType (typeof (NEProvider))]
+	[DisableDefaultCtor] // init returns nil
+	interface NEAppPushProvider {
+		[NullAllowed]
+		[Export ("providerConfiguration")]
+		NSDictionary<NSString, NSObject> ProviderConfiguration { get; }
+
+		[Async]
+		[Export ("startWithCompletionHandler:")]
+		void Start (Action<NSError> completionHandler);
+
+		[Async]
+		[Export ("stopWithReason:completionHandler:")]
+		void Stop (NEProviderStopReason reason, Action completionHandler);
+
+		[Export ("reportIncomingCallWithUserInfo:")]
+		void ReportIncomingCall (NSDictionary userInfo);
+
+		[Export ("handleTimerEvent")]
+		void HandleTimerEvent ();
+	}
+
+	[NoWatch, NoTV, Mac (11,0), iOS (14,0)]
+	[BaseType (typeof (NEDnsSettings), Name = "NEDNSOverHTTPSSettings")]
+	interface NEDnsOverHttpsSettings {
+		[NullAllowed]
+		[Export ("serverURL", ArgumentSemantic.Copy)]
+		NSUrl ServerUrl { get; set; }
+
+		[Notification]
+		[Field ("NEDNSSettingsConfigurationDidChangeNotification")]
+		NSString ConfigurationDidChangeNotification { get; }
```

Comment:
from its name it looks like it should be on `NEDnsSettings` and not on `NEDnsOverHttpsSettings`